<launch>
<!--  Set export FILE_NAME_TIME=$(date +%Y%m%d_%H%M%S) first-->

  <param name="/use_sim_time" value="false" />
  <arg name="bag_file_path" default="/home/icalab/loam_ws/src/calibration/raw_data/bag/corridor_normal/bag_$(env FILE_NAME_TIME).bag"/>
  <arg name="assemble_yr" default="/assemble_yrl" />
  <arg name="image_raw" default="/usb_cam/image_raw" />
  <arg name="tf" default="/tf" />


  <node pkg="yujin_yrl_v2_package" type="yrl_pub" name="yrl_pub" output="screen" >
    <rosparam command="load" file="$(find yujin_yrl_v2_package)/config/params.yaml"/>
  </node> 

  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
  </node>

  <node type="point_cloud2_assembler" pkg="laser_assembler" name="my_assembler">
    <remap from="cloud" to="yrl_pub/yrl_cloud"/>
    <param name="max_clouds" type="int" value="100" />
    <param name="fixed_frame" type="string" value="base_link" />
  </node>
  
  <node type="yrl2pc.py" pkg="yrl_to_cloud" name="yrl2pc" output="screen"/>
 
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find yrl_to_cloud)/config/yrl_to_cloud_rviz.rviz"/>
  
  <node name="record" pkg="rosbag" type="record" args="-O $(arg bag_file_path) $(arg assemble_yr) $(arg image_raw) $(arg tf)" output="screen"/>




</launch>

